frontend/device/android/powerd.lua

```diff
@@ -3,17 +3,48 @@ local _, android = pcall(require, "android")
 
 local AndroidPowerD = BasePowerD:new{
     fl_min = 0, fl_max = 25,
-    fl_intensity = 10,
+--    fl_intensity = 10,
 }
 
 function AndroidPowerD:frontlightIntensityHW()
-    return math.floor(android.getScreenBrightness() / 255 * self.fl_max)
+    intensity = math.floor(android.getScreenBrightness() / 255 * self.fl_max)
+    if intensity < 0 then
```

Comment:
is this because the -1? . I would like this on the backend too. So we can add a checkbox there to use system settings, like we do now.

Reply:
Yes, a checkbox in the backend to use system settings would be good. As it/was is in the frontlightwidget, then we could use the logic behind `fl_last_level`.

Reply:
I don't really understand what's the issue with fl_last_level.

On GenericController each time the activity is initialized it will pick light level from system settings. Then fl_last_level is used to override default/system settings.

Could you please explain me again why is this a problem? and where?. Just on Tolinos?

Sorry but I try to understand without testing, and that seems to break my head :/

Reply:
In my reading: fl_last_level is used by the frontlightwidget to store the last level of frontlight. So if someone just copies his koreader folder from a kobo (using frontlightwidget) to an android device, there is a chance that fl_last_level is set to a special value. -> The App would set that level on every start and you cannot change it, as there is no frontlightwidget on android.
We could extend the native dialog with a checkbox "System Light Settings". We could clear the fl_last_level, if the checkbox is checked, if it is not set, we could store the values (intensity and warmth) in fl_last_level and a new warmth_last_level. On the next start KOReader would use the Systems values or *_last_level values.

Reply:
Nope, it was added in https://github.com/koreader/koreader/commit/ae50569055a86f2672058b55dc1f1abaed8cb291 and is only used on android for now.

It is used to setup a custom fl_level at launch because, as I already said, when the activity starts it uses generic values from system settings.

Reply:
I think I will start an emulator and see what happens on your branch, because I don't get it :p

Reply:
Ok. With https://github.com/koreader/android-luajit-launcher/pull/239/commits/d2de9aab3e39119b4da4bfac1f71ea3705620229 getBrightness won't report default intensity (-1). Instead it will report system settings values, so you won't need to check intensity < 1 here.

Reply:
OK.